File: software/examples/c/example_callback.c

#include <stdio.h>

#include "ip_connection.h"
#include "bricklet_voltage_current.h"

#define HOST "localhost"
#define PORT 4223
#define UID "ABCD" // Change to your UID

// Callback function for current callback (parameter has unit mA)
void cb_current(int16_t current) {
	printf("Current: %f A\n", current/1000.0);
}

int main() {
	// Create ip connection to brickd
	IPConnection ipcon;
	if(ipcon_create(&ipcon, HOST, PORT) < 0) {
		fprintf(stderr, "Could not create connection\n");
		exit(1);
	}

	// Create device object
	VoltageCurrent vc;
	voltage_current_create(&vc, UID); 

	// Add device to ip connection
	if(ipcon_add_device(&ipcon, &vc) < 0) {
		fprintf(stderr, "Could not connect to Brick\n");
		exit(1);
	}
	// Don't use device before it is added to a connection

	// Set Period for current callback to 1s (1000ms)
	// Note: The callback is only called every second if the 
	//       current has changed since the last call!
	voltage_current_set_current_callback_period(&vc, 1000);

	// Register current callback to function cb_current
	voltage_current_register_callback(&vc, VOLTAGE_CURRENT_CALLBACK_CURRENT, cb_current);

	printf("Press ctrl+c to close\n");
	ipcon_join_thread(&ipcon); // Join mainloop of ip connection
}

File: software/examples/go/example_callback.go
package main

import (
	"fmt"
	"github.com/Tinkerforge/go-api-bindings/ipconnection"
	"github.com/Tinkerforge/go-api-bindings/voltage_current_bricklet"
)

const ADDR string = "localhost:4223"
const UID string = "XYZ" // Change XYZ to the UID of your Voltage/Current Bricklet.

func main() {
	ipcon := ipconnection.New()
	defer ipcon.Close()
	vc, _ := voltage_current_bricklet.New(UID, &ipcon) // Create device object.

	ipcon.Connect(ADDR) // Connect to brickd.
	defer ipcon.Disconnect()
	// Don't use device before ipcon is connected.

	vc.RegisterCurrentCallback(func(current int32) {
		fmt.Printf("Current: %f A\n", float64(current)/1000.0)
	})

	// Set period for current receiver to 1s (1000ms).
	// Note: The current callback is only called every second
	//       if the current has changed since the last call!
	vc.SetCurrentCallbackPeriod(1000)

	fmt.Print("Press enter to exit.")
	fmt.Scanln()
}
